pkg/resource/deploy/step_generator.go

```diff
@@ -72,10 +72,34 @@ func (sg *stepGenerator) isTargetedUpdate() bool {
 	return sg.updateTargetsOpt != nil || sg.replaceTargetsOpt != nil
 }
 
+// isTargetedForUpdate checks if a urn is target for update.
+//
+// Note: to accommodate the `--target-dependents` flag,
+// `isTargetedForUpdateOrAddIfDep` should probably be called instead.
 func (sg *stepGenerator) isTargetedForUpdate(urn resource.URN) bool {
 	return sg.updateTargetsOpt == nil || sg.updateTargetsOpt[urn]
 }
 
+// isTargetedForUpdateOrAddIfDep checks if a target should be updated, updating
+// `sg.isTargetedForUpdateOrAddIfDep` as necessary for `--target-dependents` to be
+// accurate.
+func (sg *stepGenerator) isTargetedForUpdateOrAddIfDep(urn resource.URN, dependencies []resource.URN) bool {
+	if sg.isTargetedForUpdate(urn) {
+		return true
+	}
+	// TODO; does sg.opts.TrustDependencies play a role here?
+	if !sg.opts.TargetDependents || dependencies == nil {
+		return false
+	}
+	for _, dep := range dependencies {
+		if sg.updateTargetsOpt[dep] {
+			sg.updateTargetsOpt[urn] = true
+			return true
+		}
+	}
```

Comment:
You should check the dependency graph implementation used by `--target-dependents` in the delete case and see if it considers a resource's children and a provider's associated resources to be dependents (I _think_ it does, but I'm uncertain). If it does, then this code should also check the target set for the resource's parent and provider.

Reply:
Good catch. The destroy version doesn't work that way. It instead runs the normal target destroy step generator. If it catches a resource to destroy not targeted and `--target-dependents` is not set, it errors. 

I need to check for parents and providers for the behavior to work the same way.